cEP-0024.md

```diff
@@ -0,0 +1,252 @@
+updated to remove `type/bug` label, then ask a maintainer to remove required
+label, else use `Fixes`.
+```
+
+In both cases, 'process/pending review' label will be removed and 'process/wip'
+label will be added. This will be part of `Assign label as per review state`
+plugin.
+
+### Assign Plugin
+
+When a user comments on an issue requesting for assignment to the issue,
+GitMate will assign comment author to the issue, if he/she is eligible to get
+assigned, else missing eligibility conditions will be specified as a comment.
+
+Assign plugin will support configurations for limiting number of certain
+labelled issues a member can work on, block assignment for certain users or
+certain labelled issues, difficulty level order to be followed while assigning
+issues to members so that they should work on lower difficulty level issues
+first.
+
+[corobo](https://github.com/coala/corobo) does many of these things, but it is
+based just on Gitter messages and is used only by coala, but GitMate is based
+on GitHub or GitLab, used by many other users and is configurable.
+
+**Syntax for assign command**: `@gitmate-bot assign`
+
+coala can configure this plugin as:
+
+**Block assignment of issues with labels**: `status/blocked, initiatives/*`
+
+**Block assignment of issues to users**: `gitmate-bot, coala-bot`
+
+**Maximum number of certain labelled issues a member can be assigned**:
+`difficulty/newcomer`: `1`
+
+**Difficulty Level order**:
+`difficulty/newcomer, difficulty/low, difficulty/medium`
+
+### Auto-review merge conflicts
+
+When a PR is opened or updated, check for merge conflicts in it,
+if there are any conflicts, mark the PR with a certain label (configurable) and
+post a message (configurable) as a comment on the PR.
+
+To know if PR has merge conflicts, `mergeable` property of PR in GitHub
+[API](https://developer.github.com/v3/pulls/#get-a-single-pull-request)
+and `merge_status` property of MR in GitLab
+[API](https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr) will be
+used.
+
+If `mergeable` property is False, then PR has merge conflicts, similarly if
+`merge_status` property is `cannot_be_merged`, then fastforwad merge is not
+possible, MR needs to be rebased..
+
+coala can configure this plugin as:
+
+**Label:** `needs rebase`
+
+**Comment message:** `This PR can't be merged. To merge this PR, first
+rebase locally. While rebasing, you may come across mid-rebase conflicts. For
+information regarding how to resolve mid-rebase conflicts, please check this
+[tutorial](http://gitforteams.com/resources/rebasing.html)`
+
+### Squash plugin
+
+When a user makes changes and wants to update their commit, they should use
+"git commit --amend" command, but initially some users have an habit of using
+"git commit" command, which results in creation of new commits. To fix this
+problem multiple commits need to be squashed into one commit.
+
+Newcomers often feel difficult to squash commits. So, a plugin to squash
```

Comment:
I agree newcomers find it difficult to squash commits, so did I when I made the first PR. But with this ability won't they miss the opportunity to learn `how to squash commits`?

Reply:
Maintainers will decide when this plugin can be used.